### PR TITLE
move @@type properties to type representatives

### PR DESCRIPTION
**Pull request by davidchambers**
This pull request accompanies sanctuary-js/sanctuary-type-classes#22.

### CODE REVIEW

File: test/index.js [hunk 1]
```diff
@@ -76,100 +76,122 @@ var Integer = $.NullaryType(
 );
 
 
-var MaybeTypeDict = {
-  'fantasy-land/empty': function() { return Nothing; },
-  'fantasy-land/of': function(x) { return Just(x); }
+function _Maybe(tag, value) {
```
Comment (safareli) [inline test/index.js]: Would be nice if we move this types in FL as we have them in sanctuary, sanctuary-dev and i think in sanctuary-type-classes too

Comment (davidchambers) [inline test/index.js]: I agree!

File: index.js [hunk 2]
```diff
@@ -187,29 +187,27 @@
...
-      'fantasy-land/map': function(f) { return Left(x); },
-      value: x
-    };
+  function Either(tag, value) {
+    this.isLeft = tag === 'Left';
+    this.isRight = tag === 'Right';
+    this.value = value;
   }
 
+  Either.prototype['@@type'] = 'sanctuary-def/Either';
```
Comment (davidchambers) [inline index.js]: :zap:

File: test/index.js [hunk 3]
```diff
@@ -148,9 +144,6 @@ _Either['@@type'] = 'my-package/Either';
 
 _Either['fantasy-land/of'] = Right;
 
-//  XXX: Necessary until sanctuary-type-classes#22 is merged.
-_Either.prototype['@@type'] = '';
```
Comment (safareli) [inline test/index.js]: have you missed adding of `_Either['@@type'] = ''` or is it intentional?

**Pull request merged**